### PR TITLE
[action] allow modify_services to enable data_protection 

### LINKED ISSUE #14570: modify_services - impossible to enable data_protection
### Issue Description
 ```
   modify_services(
      app_identifier: MY_APP_ID,
      team_id: MY_TEAM,
      services: {
        data_protection: "complete"
      }
    )
```

always disable data_protection. There is no way how to enable it via modify_services lane.

##### Complete output when running fastlane, including the stack trace and command used
```
[15:27:43]: -----------------------------
[15:27:43]: --- Step: modify_services ---
[15:27:43]: -----------------------------
[15:27:43]: Starting login with user '*****@icloud.com'
[15:27:44]: Successfully logged in
[15:27:45]: [DevCenter] App found '*****'
[15:27:45]: Disabling services
[15:27:45]:     Data Protection
[15:27:46]: Done! Disabled 1 services.
```

### Environment

<details><summary><b>Loaded gems</b></summary>

| Gem                                             | Version      |
| ----------------------------------------------- | ------------ |
| did_you_mean                                    | 1.3.0        |
| slack-notifier                                  | 2.3.2        |
| atomos                                          | 0.1.3        |
| CFPropertyList                                  | 3.0.0        |
| claide                                          | 1.0.2        |
| colored2                                        | 3.1.2        |
| nanaimo                                         | 0.2.6        |
| xcodeproj                                       | 1.8.1        |
| rouge                                           | 2.0.7        |
| xcpretty                                        | 0.3.0        |
| terminal-notifier                               | 2.0.0        |
| unicode-display_width                           | 1.4.1        |
| terminal-table                                  | 1.8.0        |
| plist                                           | 3.5.0        |
| public_suffix                                   | 2.0.5        |
| addressable                                     | 2.6.0        |
| multipart-post                                  | 2.0.0        |
| word_wrap                                       | 1.0.0        |
| tty-screen                                      | 0.6.5        |
| tty-cursor                                      | 0.6.0        |
| tty-spinner                                     | 0.9.0        |
| babosa                                          | 1.0.2        |
| colored                                         | 1.2          |
| highline                                        | 1.7.10       |
| commander-fastlane                              | 4.4.6        |
| excon                                           | 0.62.0       |
| faraday                                         | 0.15.4       |
| unf_ext                                         | 0.0.7.5      |
| unf                                             | 0.1.4        |
| domain_name                                     | 0.5.20180417 |
| http-cookie                                     | 1.0.3        |
| faraday-cookie_jar                              | 0.0.6        |
| fastimage                                       | 2.1.5        |
| gh_inspector                                    | 1.1.3        |
| json                                            | 2.1.0        |
| mini_magick                                     | 4.5.1        |
| multi_json                                      | 1.13.1       |
| multi_xml                                       | 0.6.0        |
| rubyzip                                         | 1.2.2        |
| security                                        | 0.1.3        |
| xcpretty-travis-formatter                       | 1.0.0        |
| dotenv                                          | 2.6.0        |
| bundler                                         | 1.17.2       |
| faraday_middleware                              | 0.12.2       |
| naturally                                       | 2.2.0        |
| simctl                                          | 1.6.5        |
| uber                                            | 0.1.0        |
| declarative                                     | 0.0.10       |
| declarative-option                              | 0.1.0        |
| representable                                   | 3.0.4        |
| retriable                                       | 3.1.2        |
| mime-types-data                                 | 3.2018.0812  |
| mime-types                                      | 3.2.2        |
| jwt                                             | 2.1.0        |
| signet                                          | 0.11.0       |
| memoist                                         | 0.16.0       |
| os                                              | 1.0.0        |
| googleauth                                      | 0.6.7        |
| httpclient                                      | 2.8.3        |
| google-api-client                               | 0.23.9       |
| google-cloud-env                                | 1.0.5        |
| google-cloud-core                               | 1.2.7        |
| digest-crc                                      | 0.4.1        |
| google-cloud-storage                            | 1.15.0       |
| emoji_regex                                     | 1.0.1        |
| forwardable                                     | 1.2.0        |
| logger                                          | 1.3.0        |
| ipaddr                                          | 1.2.2        |
| openssl                                         | 2.1.2        |
| ostruct                                         | 0.1.0        |
| date                                            | 1.0.0        |
| fileutils                                       | 1.1.0        |
| rexml                                           | 3.1.9        |
| psych                                           | 3.1.0        |
| mutex_m                                         | 0.1.0        |
| fastlane-plugin-cordova                         | 2.2.1        |
| fastlane-plugin-upgrade_super_old_xcode_project | 0.0.2        |
</details>

### PR DESCRIPTION
**Pull request by martinkasa**
Fixes enabling of services with string values other than "on". Example:
```
    modify_services(
      app_identifier: MY_APP_ID,
      team_id: MY_TEAM,
      services: {
        data_protection: "complete"
      }
    )
```

Problem was that this lane disabled service instead of enable. This commit disables all services with value "off" or false and enables other ones.

fixes #14570

### CODE REVIEW

**Comment by martinkasa** [comment]
I signed it!

On 11 Apr 2019, at 16:15, googlebot <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

📝 Please visit https://cla.developers.google.com/ to sign.

Once you've signed (or fixed any issues), please reply here (e.g. I signed it!) and we'll verify it.

________________________________
What to do if you already signed the CLA
Individual signers

  *   It's possible we don't have your GitHub username or you're using a different email address on your commit. Check your existing CLA data<https://cla.developers.google.com/clas> and verify that your email is set on your git commits<https://help.github.com/articles/setting-your-email-in-git/>.

Corporate signers

  *   Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to go/cla#troubleshoot<http://go/cla#troubleshoot> (Public version<https://opensource.google.com/docs/cla/#troubleshoot>).
  *   The email used to register you as an authorized contributor must be the email used for the Git commit. Check your existing CLA data<https://cla.developers.google.com/clas> and verify that your email is set on your git commits<https://help.github.com/articles/setting-your-email-in-git/>.
  *   The email used to register you as an authorized contributor must also be attached to your GitHub account<https://github.com/settings/emails>.

ℹ️ Googlers: Go here<https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Ffastlane%2Ffastlane%2Fpull%2F14571> for more info.

**Comment by janpio** [comment]
Unfortunately tests are failing now. Can you fix that?
Can you also explain what exactly the code changes you did do?

**Comment by martinkasa** [comment]
Before this change all values were replaced by either "on" or "off" strings. Which caused other string values as for example"complete" for data_protection feature were replaced by "off". So it was not possible to turn on services which has values other than boolean.

My change fixes this. 
- values "on", true, :on are replaced by "on"
- values "off", false and :off is replaced by "off"
- other values it keeps as they are

**Comment by martinkasa** [comment]
I have already changed email by "git config user.email " command in current repository, then pushed empty commit, but still  there is the CLA problem...

**Comment by martinkasa** [comment]
I signed it!

**Comment by janpio** [comment]
Your second commit looks good, but there is still the first one in this PR. 

You could do a "reset" of your repo to before you committed and keep your local changes, then commit them again with your now correct configuration, then force push.

Or, if this sounds scary, just create another PR with a new branch where you commit only with your correct configuration, then the CLA should be recognized instantly.

File: fastlane/spec/actions_specs/modify_services_spec.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@
                 associated_domains: 'off',
                 wallet: :on,
                 apple_pay: :off,
-                data_protection: true,
```
Comment (janpio) [inline fastlane/spec/actions_specs/modify_services_spec.rb]: This was the test if `true` is changed to `on` I guess. You should add another service to the list that is set to `true` so this behavior will still be tested after your changes.

File: fastlane/lib/fastlane/actions/modify_services.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ def self.run(params)
           require 'produce/service'
           services = params[:services]
 
-          enabled_services = services.select { |_k, v| v == true || v.to_s == 'on' }.map { |k, v| [k, 'on'] }.to_h
+          enabled_services = services.select { |_k, v| v == true || v.to_s != 'off' }.map { |k, v| [k, v == true || v.to_s == 'on' ? 'on' : v] }.to_h
```
Comment (janpio) [inline fastlane/lib/fastlane/actions/modify_services.rb]: My Ruby is to bad to understand this. Can someone explain this line in plain English what is going on here?

Comment (martinkasa) [reply fastlane/lib/fastlane/actions/modify_services.rb]: select { |_k, v| v == true || v.to_s != 'off' } -> similar to filter, it select all entries (key, value pairs) where value is true OR value converted to string is different from "off". This condition matches values: true, "on", :on, "complete", "other_string".

map { |k, v| [k, v == true || v.to_s == 'on' ? 'on' : v] } -> converts values of key-value pair to either "on" or "string". Values true, "on" and :on it converts value to "on". Values "complete", "other_string" are kept the same.

**Review by joshdholtz — APPROVED** [review]
Good find here! Thanks for the contribution ❤️💪
